genesis/src/lib.rs

```diff
@@ -58,17 +58,15 @@ impl GenesisNetwork {
         .chain(raw_block.transactions);
 
         let transactions = transactions_iter
-            .enumerate()
-            .map(|(i, raw_transaction)| {
+            .map(|raw_transaction| {
                 raw_transaction
                     // FIXME: fix underlying chain of `.sign` so that it doesn't
                     //        consume the key pair unnecessarily. It might be costly to clone
                     //        the key pair for a large genesis.
                     .sign(genesis_key_pair.clone())
```

Comment:
This FIXME is related exactly to the change you made in this PR. I guess it should be removed.

Reply:
We should also consider not consuming the key pair by-value, but taking it by-reference.

Key pairs are still non-Copy (due to `BlsSmallPublicKey` being too large and hence being boxed).

Also all of the private keys will need be boxed for `zeroize` support, so even if we ignore the public keys, it's not gonna be cheap.

IMO, there's no benefit to take them by-value, it's just an API nuisance which also has a negative performance effect

Reply:
I don't think it's currently possible to avoid clonning here. That's because of our `Signature` API design. It stores a public key inside to simplify verification later.

If we want to avoid it we need to to redisign `Signature`. That's outside of the scope of this PR and needs further investigation. I'll create an issue

Reply:
Ooh, I actually missed that. I just was thinking that it's because `iroha_crypto` consumes the keys, which is not actually the case.

Reply:
- #4199